references/product-recommendations-api/apiproxy/policies/AM-process-request-params.xml

```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<!--
+ Copyright 2021 Google LLC
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+-->
+<AssignMessage async="false" continueOnError="false" enabled="true" name="AM-process-request-params">
+    <DisplayName>AM-process-request-params</DisplayName>
+    <AssignVariable>
+        <Name>userid</Name>
+        <Value>6929470170340317899-1</Value>
```

Comment:
It's only used if a value is not provided in the test curl calls.
It's required for the AppSheet part of the Tutorial doc, since there's no way to add headers to an AppSheet data source request.
```
    <AssignVariable>
        <Name>userid</Name>
        <Value>6929470170340317899-1</Value>
        <Ref>request.header.x-userid</Ref>
    </AssignVariable>
```

Reply:
@kurtkanaskie That's the diffucult point. The Appsheet part of the Tutorial taints the reference and introduces designs that make is less valuable outside the context of the tutorial. I'd like for this reference to standalone without the implicit Appsheet dependency. Appsheet should be able to add path parameters, right? Can you move the user to the path then?

Reply:
Appsheet does not add dynamic path parameters or query params to my knowledge.

Are you suggesting one of these forms?
GET /v1/recommendations/products?userid=12345
GET /v1/recommendations/users/12345/products